okta/client.go

```diff
@@ -1470,9 +1470,43 @@ func (c *APIClient) doWithRetries(ctx context.Context, req *http.Request) (*http
 		maxRetries: c.cfg.Okta.Client.RateLimit.MaxRetries,
 	}
 	operation := func() (*http.Response, error) {
-		// Always rewind the request body when non-nil.
-		if bodyReader != nil {
-			req.Body = bodyReader()
+		if bOff.retryCount > 0 && (c.cfg.Okta.Client.AuthorizationMode == "PrivateKey" || c.cfg.Okta.Client.AuthorizationMode == "JWT") {
+			// Clear the token cache to force fresh authorization
+			// This will get a new access token and potentially a new nonce
+			c.tokenCache.Delete(AccessTokenCacheKey)
+			c.tokenCache.Delete(DpopAccessTokenNonce)
+			c.tokenCache.Delete(DpopAccessTokenPrivateKey)
+
+			headerParams := make(map[string]string)
+			queryParams := req.URL.Query()
+			req.URL.RawQuery = ""
+			auth, err := c.prepareRequest(ctx, req.URL.String(), req.Method, nil, headerParams, queryParams, url.Values{}, []formFile{})
```

Comment:
I had to use the full URL here without the query otherwise I got an error that the `htu` claim is invalid